src/atomdb/AtomDBAPITypes.h

```diff
@@ -42,32 +32,14 @@ class HandleList {
     virtual unsigned int size() = 0;
 };
 
-class RedisSet : public HandleList {
-   public:
-    RedisSet(redisReply* reply);
-    ~RedisSet();
-
-    const char* get_handle(unsigned int index);
-    unsigned int size();
-
-   private:
-    unsigned int handles_size;
-    char** handles;
-    redisReply* redis_reply;
-};
-
-class RedisStringBundle : public HandleList {
+class HandleSet {
    public:
-    RedisStringBundle(redisReply* reply);
-    ~RedisStringBundle();
-
-    const char* get_handle(unsigned int index);
-    unsigned int size();
+    HandleSet() {}
+    virtual ~HandleSet() {}
 
-   private:
-    unsigned int handles_size;
-    char** handles;
-    redisReply* redis_reply;
+    virtual unsigned int size() = 0;
+    virtual void append(void* data) = 0;
```

Comment:
I believe a more elegant solution would be to have:

```
virtual void append(shared_ptr<HandleSet> other)
```

So, inside RedisMongoDBAPITypes.cc you'd do something like:

```
void HandleSetRedis::append(shared_ptr<HandleSet> other) {
    shared_ptr<HandleSet> handle_set_redis = dynamic_cast<HandleSetRedis>(other); 
    for (auto reply : handle_set_redis->replies) {
        this->replies.push_back(reply)
        this->handles_size += reply->elements;    
    }
}
```

The extra overhead here is minimal. It's the creation (and deletion) of one extra HandleSet object per page. And the design benefits are many. Future implementations of HandleSet to other databases would have a more orthogonal behavior.

Reply:
This is pseudo-code. I didn't tried it ;-)